File: frontend/src/app/constants/payment-methods.ts
import { PaymentMethod } from '../models/order.models';

export const DELIVERY_FEE = 1500;

export interface PaymentOption {
  value: PaymentMethod;
  label: string;
}

export const PAYMENT_METHODS: PaymentOption[] = [
  { value: 'cash', label: 'Efectivo' },
  { value: 'debit', label: 'Débito' },
  { value: 'credit', label: 'Crédito' },
  { value: 'mercado_pago', label: 'Mercado Pago' },
  { value: 'voucher', label: 'Vale / Cuenta' }
];

export const PAYMENT_METHODS_DELIVERY: PaymentOption[] =
  PAYMENT_METHODS.filter(m => m.value !== 'cash');

export function paymentMethodLabel(method: string | null | undefined): string {
  if (!method) return 'Sin registrar';
  const found = PAYMENT_METHODS.find(m => m.value === method);
  return found ? found.label : method;
}

File: backend/YesChef.Core/PaymentMethods.cs
namespace YesChef.Core;

/// <summary>
/// Fuente única de verdad de los métodos de pago del sistema.
/// </summary>
public static class PaymentMethods
{
    public static readonly string[] Valid =
    {
        "cash", "debit", "credit", "mercado_pago", "voucher"
    };

    private static readonly Dictionary<string, string> Labels = new()
    {
        ["cash"] = "Efectivo",
        ["debit"] = "Débito",
        ["credit"] = "Crédito",
        ["mercado_pago"] = "Mercado Pago",
        ["voucher"] = "Vale / Cuenta"
    };

    /// <summary>Normaliza el método a minúsculas o devuelve null si es vacío.</summary>
    public static string? Normalize(string? method)
    {
        if (string.IsNullOrWhiteSpace(method))
            return null;
        return method.Trim().ToLowerInvariant();
    }

    public static bool IsValid(string? method) => Normalize(method) is not null && Valid.Contains(Normalize(method));

    public static string GetLabel(string method)
    {
        var key = Normalize(method) ?? method;
        return Labels.TryGetValue(key, out var label) ? label : key;
    }

    public static IDictionary<string, string> AllLabels => Labels;
}